request-manager/graph/grapher.go

```diff
@@ -361,26 +410,28 @@ func dependenciesSatisfied(inGraph map[string]bool, dependencies []string) bool
 	return true
 }
 
-// Returns a list of node args that aren't present in the job args map.
-func getMissingArgs(n *spec.Node, jobArgs map[string]bool) ([]string, error) {
+// checkNodeArgs checks whether all node args are present in the job args map.
+func checkNodeArgs(n *spec.Node, jobArgs map[string]bool) error {
 	missing := []string{}
 
-	// Assert that the iterable variable is present.
+	// Assert that the iterable variable is present
 	for _, each := range n.Each {
 		if each == "" {
 			continue
 		}
 		if len(strings.Split(each, ":")) != 2 { // this is malformed input
-			return missing, fmt.Errorf("in node %s: malformed input to `each:`", n.Name)
+			return fmt.Errorf("in node %s: malformed input to `each:`", n.Name)
 		}
 		iterateSet := strings.Split(each, ":")[0]
 		if !jobArgs[iterateSet] {
 			missing = append(missing, iterateSet)
 		}
 	}
 
-	// Assert that the conditional variable is present.
-	if n.If != nil {
+	// If this is a conditional node, assert that the "if" job arg is present
+	// in the job args map.
+	// Static checks assert that for conditional nodes, if != nil.
+	if n.IsConditional() {
 		if !jobArgs[*n.If] {
 			missing = append(missing, *n.If)
```

Comment:
This is a different and more specific error than below, so let's return it as such. Something like,
`fmt.Errorf("in node %s: `if: %s`: job arg %s is not set", n.Name, n.If. nIf")`